api/src/test/java/org/motechproject/nms/api/osgi/UserControllerBundleIT.java

```diff
@@ -67,9 +117,109 @@ private void setupData() {
+        serviceUsageDataService.create(serviceUsage);
+    }
+
+    private void createFLWWithLanguageFullServiceUsageAndCappedService() {
+        cleanAllData();
+
+        Language language = new Language("English", 10);
+        languageDataService.create(language);
+
+        FrontLineWorker flw = new FrontLineWorker("Frank Llyod Wright", "0000000000");
+        flw.setLanguage(language);
+        frontLineWorkerService.add(flw);
+
+        language = new Language("Papiamento", 99);
+        languageDataService.create(language);
+
+        CircleLanguage circleLanguage = new CircleLanguage("AA", language);
+        circleLanguageDataService.create(circleLanguage);
+
+        ServiceUsageCap serviceUsageCap = new ServiceUsageCap(null, Service.MOBILE_KUNJI, 3600);
+        serviceUsageCapDataService.create(serviceUsageCap);
+
+        ServiceUsage serviceUsage = new ServiceUsage(flw, Service.MOBILE_KUNJI, 1, 1, 1, DateTime.now());
+        serviceUsageDataService.create(serviceUsage);
+    }
+
+    private void createFLWWithLanguageFullUsageOfBothServiceUncapped() {
+        cleanAllData();
+
+        Language language = new Language("English", 10);
+        languageDataService.create(language);
+
+        FrontLineWorker flw = new FrontLineWorker("Frank Llyod Wright", "0000000000");
+        flw.setLanguage(language);
+        frontLineWorkerService.add(flw);
+
+        language = new Language("Papiamento", 99);
+        languageDataService.create(language);
+
+        CircleLanguage circleLanguage = new CircleLanguage("AA", language);
+        circleLanguageDataService.create(circleLanguage);
+
+        ServiceUsage serviceUsage = new ServiceUsage(flw, Service.MOBILE_KUNJI, 1, 1, 1, DateTime.now());
+        serviceUsageDataService.create(serviceUsage);
+
+        // Academy doesn't have a welcome prompt
+        serviceUsage = new ServiceUsage(flw, Service.MOBILE_ACADEMY, 1, 1, 0, DateTime.now());
+        serviceUsageDataService.create(serviceUsage);
+
+        ServiceUsageCap serviceUsageCap = new ServiceUsageCap(null, Service.MOBILE_KUNJI, 10);
+        serviceUsageCapDataService.create(serviceUsageCap);
+    }
+
+    private void createCircleWithLanguage() {
+        cleanAllData();
+        Language language = new Language("Papiamento", 99);
+        languageDataService.create(language);
+
+        CircleLanguage circleLanguage = new CircleLanguage("AA", language);
+        circleLanguageDataService.create(circleLanguage);
+    }
+
     @Test
-    public void testUserRequest() throws IOException, InterruptedException {
-        setupData();
+    public void testKilkariUserRequest() throws IOException, InterruptedException {
+        createKilkariTestData();
+
         HttpGet httpGet = new HttpGet(String.format("http://localhost:%d/api/kilkari/user?callingNumber=0000000001&operator=OP&circle=AA&callId=0123456789abcde", TestContext.getJettyPort()));
 
         httpGet.addHeader("Authorization",
```

Comment:
Now that a better version of SimpleHttpClient is available on platform, we should use
```java
assertTrue(SimpleHttpClient.execHttpRequest(httpGet, "...", ADMIN_USERNAME, ADMIN_PASSWORD));
```
instead of adding the AUTH headers to the request